Fix artificial view update size limit

psarna opened the pull request:
The series which split the view update process into smaller parts
accidentally put an artificial 10MB limit on the generated mutation
size, which is wrong - this limit is configurable for users,
and, what's more important, this data was already validated when
it was inserted into the base table. Thus, the limit is lifted.
  
The series comes with a cql-pytest which failed before the fix and succeeds now. This bug is also  covered by `wide_rows_test.py:TestWideRows_with_LeveledCompactionStrategy.test_large_cell_in_materialized_view` dtest, but it needs over a minute to run, as opposed to cql-pytest's <1 second. 

Fixes #9047

Tests: unit(release), dtest(wide_rows_test.py:TestWideRows_with_LeveledCompactionStrategy.test_large_cell_in_materialized_view)

nyh commented on test/cql-pytest/test_materialized_view.py:
```diff
@@ -0,0 +1,45 @@
+        cql.execute(f"CREATE MATERIALIZED VIEW {test_keyspace}.{mv} AS SELECT * FROM {table} WHERE c IS NOT NULL PRIMARY KEY (c,p)")
+        try:
+            retrieved_row = False
+            for _ in range(50):
```
"5 seconds is more than enough time for the view build to finish" - famous last words :-(
I guess we can increase it if it starts failing on some busy setup in debug mode.

psarna replied:
My hope is that in debug build, executing the select statement could itself take an order of magnitude longer than the 0.1 sleep time. I'm crossing my fingers that it would take long enough.

nyh commented on test/cql-pytest/test_materialized_view.py:
```diff
@@ -0,0 +1,45 @@
+
+from util import new_test_table, unique_name
+
+# Test that building an index with a large value succeeds. Regression test
```
This test is specific to "view building", i.e., adding a view on a table with pre-existing data. Is the bug specific to view building, or also exists when adding a row to a table which already has a MV enabled?

nyh replied:
By the way, the reference to "index" is wrong - it's a materialized view in this test, not an index.

psarna replied:
Actually, the bug is in both paths (regular write and building). The dtest was about view building and that's where I took the inspiration, but it's a good idea to add another case for the view path. It's convenient that on a 1-node cluster each view is local, so I can do away without waiting for eventual consistency.

nyh commented:
Oh, @psarna I think you forgot to add Fixes #9047 ?

psarna commented:
v2:
 - fixed mentioning of "index" in the comment when it was a fact a view
 - added a mirror test case for updating an existing materialized view (as opposed to building the view from scratch)

psarna commented:
oh wait, these tests should not be `scylla_only` - let me fix that too

psarna commented:
v3:
 - tests are no longer scylla_only, there was no reason to skip verifying against cassandra
